packages/show-format/src/trajectory.ts

```diff
@@ -289,51 +291,155 @@ export function createTimedBezierCurve(
 }
 
 /**
- * Splits the given curve at the given fraction (aka `t`) into two segments
- * using de Casteljau's algorithm.
+ * Converts a `TimedBezierCurve` object to a `TrajectorySegment` object.
  *
- * @param segment The segment to split.
+ * Note: the first control point of the curve is lost during the transition, because
+ * in the `TrajectorySegment` representation it is defined by the preceding segment.
+ *
+ * @param curve The curve to convert.
+ *
+ * @returns The trajectory segment corresponding to the curve.
+ *
+ * @throws `Error` if the curve has less than 2 control points.
+ */
+export function timedBezierCurveToTrajectorySegment(
+  curve: TimedBezierCurve
+): TrajectorySegment {
+  const { duration, startTime, points } = curve;
+  if (points.length < 2) {
+    throw new Error('The curve must have at least 2 control points.');
+  }
+
+  // [end time of segment, endpoint of segment, additional control points ]
+  // splice() first index is 1, because the first point is defined by the
+  // previous segment (endpoint).
+  return [startTime + duration, points.at(-1)!, points.slice(1, -1)];
+}
+
+/**
+ * Splits the given curve at the given fraction (aka `t`) into two segments.
+ *
+ * @param curve The curve to split.
  * @param fraction The fraction at which to split the segment.
  *
- * @returns The two resulting trajectory segments.
+ * @returns The two resulting curves.
+ *
+ * @throws `Error` if the fraction is not in the [0, 1] interval.
  */
 export function splitTimedBezierCurve(
-  segment: TimedBezierCurve,
+  curve: TimedBezierCurve,
   fraction: number
-): [TrajectorySegment, TrajectorySegment] {
+): [TimedBezierCurve, TimedBezierCurve] {
   if (fraction < 0 || fraction > 1) {
     throw new Error('fraction must be in the [0, 1] interval.');
   }
 
-  const { startTime, duration, points } = segment;
+  const { startTime, duration, points } = curve;
   if (points.length < 2) {
     throw new Error('The curve must have at least 2 control points.');
   }
 
   if (fraction === 0) {
     return [
-      [startTime, points[0], []],
-      [startTime + duration, points.at(-1)!, points.slice(1, -1)],
+      { startTime, duration: 0, points: [points[0], points[0]] },
+      { startTime, duration, points },
     ];
   } else if (fraction === 1) {
     return [
-      [startTime + duration, points.at(-1)!, points.slice(1, -1)],
-      [startTime + duration, points.at(-1)!, []],
+      { startTime, duration, points },
+      {
+        startTime: startTime + duration,
+        duration: 0,
+        points: [points[0], points.at(-1)!],
```

Comment:
Shouldn't this be `points.at(-1)` twice?

Reply:
I think you're right. Thanks for catching it, I should have noticed it.

I'll fix it today and then check whether tests pass if I remove the `pre` and `post` trajectories from them (they should pass like that as well).

Reply:
Fixed.